### PR TITLE
Update summaries to be indented based on the featureColumnsWidth in all cases and update Hierarchical Grid. 

### PR DESCRIPTION
**Pull request by skrustev**
#4896

### Additional information (check all that apply):
 - [x] Bug fix
 - [ ] New functionality
 - [ ] Documentation
 - [ ] Demos
 - [ ] CI/CD

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [ ] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [ ] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [ ] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.integration.spec.ts [hunk 1]
```diff
@@ -426,8 +426,8 @@ describe('IgxHierarchicalGrid Integration', () => {
             expect(summaryRow.children[0].offsetWidth).toEqual(expander.nativeElement.offsetWidth);
             expect(summaryRow.children[1].tagName.toLowerCase()).toEqual('igx-display-container');
 
-            // there should be indentation of the summaries
-            expect(summaryRow.children[0].className.indexOf(INDENT_LEVEL_CLASS) !== -1).toBe(true);
+             // there should be indentation of the summaries
+             expect(summaryRow.children[0].className.indexOf(INDENT_LEVEL_CLASS) !== -1).toBe(false);
```
Comment (MayaKirova) [inline projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.integration.spec.ts]: This class INDENT_LEVEL_CLASS - igx-grid__row-indentation--level-1, does not seem to be used anymore. It appears that the summaries indent is handled by another class - igx-grid__summaries-patch and its width is bound to summariesMargin. So checks that depend on old class should be changed.

File: projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.integration.spec.ts [hunk 2]
```diff
@@ -450,6 +450,42 @@ describe('IgxHierarchicalGrid Integration', () => {
...
+                .filter((child: any) => child.tagName.toLowerCase() !== 'igx-display-container');
+            const childIndentCombinedWidth = childIndentChildren.reduce((prev: any, cur: any) => prev + cur.offsetWidth, 0);
+
+            expect(summaryRow.children.length).toEqual(2);
+            expect(summaryRow.children[0].tagName.toLowerCase()).toEqual('div');
+            expect(summaryRow.children[1].tagName.toLowerCase()).toEqual('igx-display-container');
+            expect(childIndentCombinedWidth).toEqual(expander.nativeElement.offsetWidth);
+
+            // there should be indentation of the summaries
+            expect(summaryRow.children[0].className.indexOf(INDENT_LEVEL_CLASS) !== -1).toBe(false);
```
Comment (MayaKirova) [inline projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.integration.spec.ts]: INDENT_LEVEL_CLASS checks should be updated.

File: projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.integration.spec.ts [hunk 3]
```diff
@@ -450,6 +450,42 @@ describe('IgxHierarchicalGrid Integration', () => {
...
+            const rootCheckbox =  hierarchicalGrid.headerCheckboxContainer;
+            const rootSummaryRow = hierarchicalGrid.summariesRowList.first.nativeElement;
+            const rootIndentChildren = Array.from(rootSummaryRow.children)
+                .filter((child: any) => child.tagName.toLowerCase() !== 'igx-display-container');
+            const indentCombinedWidth = rootIndentChildren.reduce((prev: any, cur: any) => prev + cur.offsetWidth, 0);
+
+            expect(rootSummaryRow.children.length).toEqual(2);
+            expect(rootIndentChildren.length).toEqual(1);
+            expect(indentCombinedWidth).toEqual(rootExpander.nativeElement.offsetWidth + rootCheckbox.nativeElement.offsetWidth);
+            expect(rootSummaryRow.children[0].className.indexOf(INDENT_LEVEL_CLASS) !== -1).toBe(false);
```
Comment (MayaKirova) [inline projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.integration.spec.ts]: INDENT_LEVEL_CLASS checks should be updated.

File: projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.integration.spec.ts (hunk 1 above)
Comment (skrustev) [reply projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.integration.spec.ts]: The check is intentional in this is case so it should not display indentation level element. 
It is still present in the summary row component.

**Pull request merged**